### ISSUE TITLE
[Bug]: Posting to / with `useFetcher().Form` tries to post to index route, instead of the root route.

### ISSUE THREAD

**Issue by DaCurse** [remix-run/remix#1800]
### Which Remix packages are impacted?

- [X] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [ ] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

1.1.3

### What version of Node are you using? Minimum supported version is 14.

v17.1.0

### Steps to Reproduce

Repository: https://github.com/DaCurse/remix-issue-1800

`useFetcher().Form` with `action="/"` or no action when rendered in root route, tries to post to `routes/index` instead of to the root route. The fetchers post just fine if `routes/index.tsx` is deleted.

### Expected Behavior

A form with `action="/"` or no `action` that is rendered in the root route should post to the root route, and not to the index route

### Actual Behavior

The form tries to post to the index route (`routes/index`), and fails if it doesn't export an action.
```
Uncaught (in promise) Error: Route "routes/index" does not have an action, but you are trying to submit to it. To fix this, please add an `action` function to the route
    at callAction (transition.js:858:11)
```
If `routes/index.tsx` doesn't exist, it posts to the root route as expected.

**Comment by webtree22** [comment]
I have exact same problem. is there any way around  ?

**Comment by confix** [comment]
The behavior seems to be caused by https://github.com/remix-run/remix/blob/712ca2b5b08eb3200b5cb1bbf29e8faa77f556d1/packages/remix-react/transition.ts#L509.

It was working as expected when using `<Form />` instead of a `useFetcher().Form`.

**Comment by andresgutgon** [comment]
Oh man, this is [explained here](https://github.com/remix-run/remix/pull/284#issuecomment-924469956) I had the same problem:

## NOT working
![image](https://user-images.githubusercontent.com/49499/152958412-8eb38bfd-dc5d-46e0-b87a-3e3e200cc092.png)

## Working
![image](https://user-images.githubusercontent.com/49499/152958473-e22aab91-e6a4-4154-b69b-c38397e757b8.png)

You have to tell remix is the `?index` action what you want to call

@ryanflorence is this explained somewhere in the docs? 
Maybe put [your explanation here](https://github.com/remix-run/remix/pull/284#issuecomment-924469956) and make it part of `The Remix Way ®`

**Comment by DaCurse** [comment]
> Oh man, this is [explained here](https://github.com/remix-run/remix/pull/284#issuecomment-924469956) I had the same problem:
> 
> ## NOT working
> ![image](https://user-images.githubusercontent.com/49499/152958412-8eb38bfd-dc5d-46e0-b87a-3e3e200cc092.png)
> 
> ## Working
> ![image](https://user-images.githubusercontent.com/49499/152958473-e22aab91-e6a4-4154-b69b-c38397e757b8.png)
> 
> You have to tell remix is the `?index` action what you want to call
> 
> @ryanflorence is this explained somewhere in the docs? 
> Maybe put [your explanation here](https://github.com/remix-run/remix/pull/284#issuecomment-924469956) and make it part of `The Remix Way ®`

It's not the same problem. I'm talking about posting to the *root* route, not to the index route.

**Comment by andresgutgon** [comment]
> It's not the same problem. I'm talking about posting to the root route, not to the index route.

Ahhh, sorry for the noise

Anyway I learn something new with the `?index` way, I hope someone can find it helpful. And again sorry for the misunderstanding

**Comment by brophdawg11** [comment]
Thanks for the reproduction @DaCurse!  This doesn't seem to be reproducible in the latest version - I think it was fixed by #2319.  I'm going to close this out, but please feel free to re-open if you are still running into issues.
